src/libtiled/mapreader.cpp

```diff
@@ -834,6 +873,48 @@ void MapReaderPrivate::readImageLayerImage(ImageLayer &imageLayer)
     xml.skipCurrentElement();
 }
 
+inline void MapReaderPrivate::readObjectData(MapObject *object)
+{
+    while (xml.readNextStartElement()) {
+        if (xml.name() == QLatin1String("properties")) {
+            object->mergeProperties(readProperties());
+        } else if (xml.name() == QLatin1String("polygon")) {
+            object->setPolygon(readPolygon());
+            object->setShape(MapObject::Polygon);
+        } else if (xml.name() == QLatin1String("polyline")) {
+            object->setPolygon(readPolygon());
+            object->setShape(MapObject::Polyline);
+        } else if (xml.name() == QLatin1String("ellipse")) {
+            xml.skipCurrentElement();
+            object->setShape(MapObject::Ellipse);
+        } else if (xml.name() == QLatin1String("text")) {
+            object->setTextData(readObjectText());
+            object->setShape(MapObject::Text);
+        } else {
+            readUnknownElement();
+        }
+    }
+}
+
+MapObject *MapReaderPrivate::readTemplate()
+{
+    Q_ASSERT(xml.isStartElement() && xml.name() == QLatin1String("template"));
+    const QXmlStreamAttributes atts = xml.attributes();
+    const QString name = atts.value(QLatin1String("name")).toString();
+//    const unsigned gid = atts.value(QLatin1String("gid")).toUInt();
+    const qreal width = atts.value(QLatin1String("width")).toDouble();
+    const qreal height = atts.value(QLatin1String("height")).toDouble();
+    const QString type = atts.value(QLatin1String("type")).toString();
+
+    const QSizeF size(width, height);
+
+    MapObject *object = new MapObject(name, type, {0, 0}, size);
+
+    readObjectData(object);
```

Comment:
Alright, I didn't understand what you meant, I thought you referred to the template group as template.

So just to be clear the format will look like this
```
<templategroup ...>
 <template ...>
  <object ...>
    ...
  </object ...>
 </template ...>

 <template ...>
  <object ...>
    ...
  </object ...>
 </template>
</templategroup>
```
This looks more effective, I will update it and the template class makes more sense now.

And for writing, is there a better way other than creating a new function just to prevent writing the id and the coordinates? Does sending a parameter to tell it whether to write them or not be appropriate?